incubator/hnc/internal/validators/object.go

```diff
@@ -275,16 +276,19 @@ func (o *Object) handleInherited(op k8sadm.Operation, newSource, oldSource strin
 	return deny(metav1.StatusReasonInternalError, "unknown operation: "+string(op))
 }
 
-// isDeletingNS returns true if the namespace of the object is already being deleted, based on the
-// in-memory forest.
-func (o *Object) isDeletingNS(inst *unstructured.Unstructured) bool {
-	o.Forest.Lock()
-	defer o.Forest.Unlock()
-	ns := o.Forest.Get(inst.GetNamespace())
-	if ns == nil {
-		return false
+// validateDeletingNS validates if the namespace of the object is already being deleted
+func (o *Object) validateDeletingNS(ctx context.Context, inst *unstructured.Unstructured) error {
```

Comment:
Errors are usually there to define exceptional cases, but in this case I wouldn't call "the namespace is not being deleted" an exception. So I think I'd go back to calling this `isDeletingNS` but allow it to return both a boolean and an error; we only return an error if the call to `Get` fails and we can't tell whether the namespace is being deleted or not.